src/types/from_node.rs

```diff
@@ -0,0 +1,357 @@
+use std::time::Duration;
+
+use anyhow::anyhow;
+use compositor_common::{
+    scene::{
+        self,
+        builtin_transformations::{
+            self, tiled_layout::TiledLayoutSpec, BuiltinSpec, FixedPositionLayoutSpec,
+        },
+        shader,
+        text_spec::{self, TextSpec},
+        transition, NodeSpec, MAX_NODE_RESOLUTION,
+    },
+    util::colors::{self, RGBAColor},
+};
+
+use super::node::*;
+use super::util::*;
+
+impl TryFrom<Node> for NodeSpec {
+    type Error = anyhow::Error;
+
+    fn try_from(node: Node) -> Result<Self, Self::Error> {
+        let params = match node.params {
+            NodeParams::WebRenderer(node) => node.into(),
+            NodeParams::Shader(node) => node.into(),
+            NodeParams::Image(node) => node.into(),
+            NodeParams::Text(node) => node.try_into()?,
+            NodeParams::Transition(node) => node.try_into()?,
+            NodeParams::TransformToResolution(node) => scene::NodeParams::Builtin {
+                transformation: node.try_into()?,
+            },
+            NodeParams::FixedPositionLayout(node) => scene::NodeParams::Builtin {
+                transformation: node.try_into()?,
+            },
```

Comment:
With a new API layer, `transformation` is useless. It was only used for the correct JSON type, but we can use `scene::NodeParams::Builtin(BuiltinSpec)` now

Reply:
yes it is, but I don't want to change anything in API (both http and renderer) in this PR if I don't have to